pkg/resource/configurator.go

```diff
@@ -52,16 +52,21 @@ func NewObjectMetaConfigurator(t runtime.ObjectTyper) *ObjectMetaConfigurator {
 }
 
 // Configure the supplied Managed resource's object metadata.
-func (c *ObjectMetaConfigurator) Configure(_ context.Context, cm Claim, cs NonPortableClass, mg Managed) error {
-	mg.SetNamespace(cs.GetNamespace())
+func (c *ObjectMetaConfigurator) Configure(_ context.Context, cm Claim, _ Class, mg Managed) error {
 	mg.SetGenerateName(fmt.Sprintf("%s-%s-", cm.GetNamespace(), cm.GetName()))
 	if meta.GetExternalName(cm) != "" {
 		meta.SetExternalName(mg, meta.GetExternalName(cm))
 	}
-	// TODO(negz): Don't set this potentially cross-namespace owner reference.
-	// We probably want to use the resource's reclaim policy, not Kubernetes
-	// garbage collection, to determine whether to delete a managed resource
-	// when its claim is deleted per https://github.com/crossplaneio/crossplane/issues/550
+
+	// TODO(negz): Avoid setting this owner reference? Kubernetes specifies that
```

Comment:
Do you know what happens if we try and set this owner reference?

Reply:
I haven't looked at this for some time, but as far as I can tell it works. Currently we almost always dynamically provision a managed resource in a separate namespace from the resource claim, yet deleting the resource claim automatically deletes the managed resource (via owner reference based garbage collection).

What's interesting is that I have no idea _how_ this works - I'm not sure how the garbage collector figures out what to delete without knowing the owner's namespace.

Reply:
I am going to dive into this a little bit more, but I don't think my investigation needs to block here :)